xray/plot/plot.py

```diff
@@ -452,6 +452,8 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
             # pcolormesh
             kwargs['extend'] = cmap_params['extend']
             kwargs['levels'] = cmap_params['levels']
+            if 'colors' in kwargs:
```

Comment:
Yes, I think that's probably a better way to provide explicit color(s) than overloading cmap. We can raise an error is colors is provided without the levels arguments.